src/main/java/org/trips/service_framework/services/CmsService.java

```diff
@@ -189,14 +188,67 @@ public Sku createSku(SkuAttributes skuAttributes) {
 
     public Sku getOrCreateSku(SkuAttributes attributes) {
         try {
-            CmsSkuResponse response = getSkuByAttributes(attributes);
-            return response.getData().getSearchSkus().get(0);
+            return getSkuByAttributes(attributes);
         } catch (CmsException cmsException) {
             log.info("SKU not found. Creating new SKU for attributes {}", attributes);
             return createSku(attributes);
         }
     }
 
+    // This method returns a list of alike SKUs based on the attributes provided.
+    public List<Sku> getSkusByAttributes(SkuAttributes attributes, boolean getLiteResponse) {
+        Map<String, Object> requestParams = cmsHelper.getQueryForSkusSearch(attributes);
+
+        log.info("Searching for SKU(s) by attributes from CMS, Payload: {}", requestParams);
+        String resourcePath = getLiteResponse ? "searchSkuLite.graphql" : "searchSku.graphql";
+        CmsSkuResponse response = skuSearchHelper("SearchSkus", resourcePath, requestParams);
+
+        if (Objects.isNull(response) || Objects.isNull(response.getData())) {
+            throw new CmsException("SKU search by attributes returned null response");
+        }
+
+        if (CollectionUtils.isEmpty(response.getData().getSearchSkus())) {
+            throw new CmsException("No SKU found for the given sku attributes");
+        }
+
+        return response.getData().getSearchSkus();
+    }
+
+    public List<Sku> skuSearch(Map<String, String> body) {
+        String searchType = Optional.ofNullable(body.get("search_type"))
```

Comment:
if all the possible request body params are defined then lets use a defined Java class instead of Map. If using java class then this validation can be done with the help of spring validation using @Validate.